integration/epochs/epoch_qc_test.go

```diff
@@ -73,15 +67,24 @@ func (s *Suite) TestEpochQuorumCertificate() {
 
 		// create account on emualted chain
 		address, err := s.blockchain.CreateAccount([]*sdk.AccountKey{key}, []sdktemplates.Contract{})
-		require.NoError(s.T(), err)
+		s.Require().NoError(err)
 
 		client := epochs.NewQCContractClient(zerolog.Nop(), s.emulatorClient, nodeID, address.String(), 0, s.qcAddress.String(), signer)
-		require.NoError(s.T(), err)
+		s.Require().NoError(err)
 
 		local := &modulemock.Local{}
 		local.On("NodeID").Return(nodeID)
 
-		vote := hotstuffmodel.VoteFromFlow(nodeID, unittest.IdentifierFixture(), 0, unittest.SignatureFixture())
+		blockID := unittest.IdentifierFixture()
+		view := uint64(rand.Uint32())
+		voteMessage := hotstuffver.MakeVoteMessage(view, blockID)

```

Comment:
We are generating a different `blockID` and `view` within the loop over the set of nodes. This means each node will submit a **different** `voteMessage`. 

This will result in the QC aggregation process failing because the smart contract requires that a quorum of nodes submit the **same** `voteMessage` -- this is how we guarantee that we are able to produce a valid quorum certificate at the end of the process. ⚠️  If the vote signatures are over different messages, we cannot aggregate them!